changelog.d/10086.doc

```diff
@@ -0,0 +1 @@
+Add initial infrastructure for rendering Synapse documentation with mdbook.
```

Comment:
Should we put a link here to the rendered version or maybe we should add it to the release notes?

Reply:
Release notes sounds good - we'll need to flip the switch on the GitHub Pages settings once we've merged anyhow.